dap/variables.go

```diff
@@ -154,7 +169,59 @@ func execOpVars(exec *pb.ExecOp, refs *variableReferences) dap.Variable {
 	}
 }
 
+func (f *frame) fillVarsFromResult(ctx context.Context, ref gateway.Reference, refs *variableReferences) {
+	f.scopes = append(f.scopes, dap.Scope{
+		Name:             "File Explorer",
+		PresentationHint: "locals",
+		VariablesReference: refs.New(func() []dap.Variable {
+			return fsVars(ctx, ref, "/", refs)
+		}),
+		Expensive: true,
+	})
+}
+
+func fsVars(ctx context.Context, ref gateway.Reference, path string, vars *variableReferences) []dap.Variable {
+	files, err := ref.ReadDir(ctx, gateway.ReadDirRequest{
+		Path: path,
+	})
+	if err != nil {
+		return []dap.Variable{
+			{
+				Name:  "error",
+				Value: err.Error(),
+			},
+		}
+	}
+
+	paths := make([]dap.Variable, len(files))
+	for i, file := range files {
+		fs := dap.Variable{
+			Name:  file.Path,
+			Value: statf(file),
+		}
+
+		if file.IsDir() {
+			fs.Name += "/"
+			fs.VariablesReference = vars.New(func() []dap.Variable {
+				newpath := filepath.Join(path, file.Path)
+				return fsVars(ctx, ref, newpath, vars)
+			})
+		}
+		paths[i] = fs
+	}
+	return paths
+}
+
+func statf(st *types.Stat) string {
+	mode := fs.FileMode(st.Mode)
+	modTime := time.Unix(0, st.ModTime).UTC()
+	return fmt.Sprintf("%s\t%d\t%d\t%s", mode, st.Uid, st.Gid, modTime.Format("Jan 2 15:04:05 2006"))
+}
```

Comment:
The tabs might be hit-or-miss depending on the client... 🤔

<img width="427" height="462" alt="image" src="https://github.com/user-attachments/assets/86d06da4-2e35-458b-b881-d6607aeaa197" />

Reply:
Hm... I don't think the tabs are likely a good idea. I'll look into another way of representing this.